sklearn/preprocessing/tests/test_discretization.py

```diff
@@ -172,5 +177,43 @@ def test_numeric_stability():
     # Test up to discretizing nano units
     for i in range(1, 9):
         X = X_init / 10**i
-        Xt = KBinsDiscretizer(n_bins=2).fit_transform(X)
+        Xt = KBinsDiscretizer(n_bins=2, encode='ordinal').fit_transform(X)
         assert_array_equal(Xt_expected, Xt)
+
+
+def test_invalid_encode_option():
+    est = KBinsDiscretizer(n_bins=[2, 3, 3, 3], encode='invalid-encode')
+    assert_raises(ValueError, est.fit, X)
+
+
+def test_encode_options():
+    # test valid encode options through comparison
+    est = KBinsDiscretizer(n_bins=[2, 3, 3, 3],
+                           encode='ordinal').fit(X)
+    Xt_1 = est.transform(X)
+    est = KBinsDiscretizer(n_bins=[2, 3, 3, 3],
+                           encode='onehot-dense').fit(X)
+    Xt_2 = est.transform(X)
+    assert not sp.issparse(Xt_2)
+    assert_array_equal(OneHotEncoder(n_values=[2, 3, 3, 3], sparse=False)
+                       .fit_transform(Xt_1), Xt_2)
+    assert_raises(ValueError, est.inverse_transform, Xt_2)
+    est = KBinsDiscretizer(n_bins=[2, 3, 3, 3],
+                           encode='onehot').fit(X)
+    Xt_3 = est.transform(X)
+    assert sp.issparse(Xt_3)
+    assert_array_equal(OneHotEncoder(n_values=[2, 3, 3, 3], sparse=True)
+                       .fit_transform(Xt_1).toarray(),
+                       Xt_3.toarray())
+    assert_raises(ValueError, est.inverse_transform, Xt_3)
+
+
+def test_one_hot_encode_with_ignored_features():
+    est = KBinsDiscretizer(n_bins=3, ignored_features=[1, 2],
+                           encode='onehot-dense').fit(X)
+    Xt_1 = est.transform(X)
+    Xt_2 = np.array([[1, 0, 0, 1, 0, 0, 1.5, -4],
```

Comment:
Nit: You don't need to have `Xt_2` be an array. A nested list should suffice.

I would also rename `Xt_2 -> Xt_expected` and `Xt_1 -> Xt`.